UniquenessValidator exclude itself when PK changed, 4.2 backport #23581, fix #23399

diego-silva opened the pull request:
This is a backport from PR #23581 that fixes the issue reported on #23399 for Rails 4.2.

> When changing the PK for a record which has a uniqueness validation on
> some other attribute, Active Record should exclude itself from the
> validation based on the PK value stored on the DB (id_was) instead of
> its new value (id).

diego-silva committed:
UniquenessValidator exclude itself when PK changed

When changing the PK for a record which has a uniqueness validation on
some other attribute, Active Record should exclude itself from the
validation based on the PK value stored on the DB (id_was) instead of
its new value (id).

sgrif commented:
@diego-silva Can you rebase this? I'm unable to do this for you manually, unfortunately

diego-silva commented:
@sgrif sorry for the delay on this one, I had rebased it but the tests were failing even on master, and then I lost track of this PR…

I've just rebased it and all tests are green, should we merge this?

kamipo commented:
Closing since Rails 4.2 only receives security fixes now.
